knimin/templates/pm_plate_map.html

```diff
@@ -0,0 +1,358 @@
+<!-- The Plate Mapper module -->
+<!-- Plate map -->
+
+{% extends logged_in_index.html %}
+{% block head %}
+{% from json import dumps %}
+{% autoescape None %}
+<script type="text/javascript" src="/static/js/labmin.js"></script>
+<script type="text/javascript" src="/static/vendor/js/jquery.validate.min.js"></script>
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css">
+<style>
+  p {
+    font-weight: bold;
+  }
+  button {
+    padding: 5px 10px 5px 10px;
+    text-decoration: none;
+    font-size: 80%;
+  }
+  #plate_map {
+    width: 100%;
+    table-layout: fixed;
+    /* border-spacing: 0; */
+    border: 1px solid black;
+    border-collapse: collapse;
+  }
+  #plate_map td {
+    text-align: center;
+    vertical-align: middle;
+    border: 1px solid black;
+    margin: 0;
+    padding: 0;
+    background-color: rgba(255,247,188,0.75);
+  }
+  #plate_map td.head {
+    font-size: 80%;
+    padding-top: 1px;
+    padding-bottom: 1px;
+  }
+  input.cell {
+    display: block;
+    text-align: center;
+    padding: 0;
+    margin: 0;
+    border: 0;
+    width: 100%;
+    border-radius: 0;
+    line-height: 2.5;
+  }
+  input.cell:focus {
+    outline: none;
+    /*border-color: #719ECE;*/
+    box-shadow: inset 0 0 5px rgb(44,127,184); /*#719ECE;*/
+    background-color: white;
+  }
+  input[type="text"] {
+    border-radius: 0;
+  }
+  .chosen-container-single .chosen-single {
+    /* height: 30px;
+    border-radius: 3px; */
+    height: 25px;
+    border-radius: 0;
+    background-image: none;
+    border: 1px solid black;
+    margin-bottom: 1px;
+    font-weight: normal;
+    font-size: 90%;
+  }
+  .chosen-container-single .chosen-results {
+    font-weight: normal;
+    font-size: 90%;
+  }
+  .chosen-choices {
+      border: 1px solid #ccc;
+      border-radius: 4px;
+      min-height: 34px;
+      padding: 6px 12px;
+  }
+</style>
+<script type="text/javascript">
+
+  // Retrieve data
+  var target = {{dumps(target)}};                 // Target object: sample, primer, barseq, etc.
+  var id = {{dumps(id)}};                         // Plate ID (0 for creating new plate)
+  var info = {{dumps(info)}};                     // Plate info: name, email, created_on, notes
+  var type = {{dumps(type)}};                     // Plate type: name, cols, rows, notes
+  var layout = {{dumps(layout)}};                 // Plate layout: sample_id, col, row, name, notes
+  var currentuser = {{dumps(currentuser)}};       // Current user
+
+  // Pre-process data
+  // The coordinate of a well is represented by a scalar "col_row" to facilitate searching
+  var well2id = {};
+  for (var i = 0; i < layout.length; i++) {
+    well2id[layout[i]['col'].toString() + "_" + layout[i]['row'].toString()] = i;
+  }
+
+  $(document).ready(function() {
+
+    // Render display items
+    $(".chosen-select").chosen();
+    $("#plated_on").datepicker();
+    $.ajax({
```

Comment:
This is an AJAX call. Although it does not appear to be necessary at this moment, but eventually these dropdown menus will be dynamically populated, as the user can create new entries.